src/main/java/com/marklogic/kafka/connect/source/AbstractRowBatchBuilder.java

```diff
@@ -84,4 +91,23 @@ private void configureBatchView(Map<String, Object> parsedConfig, RowBatcher<?>
     private void onFailureHandler(RowBatchFailureListener.RowBatchFailureEvent batch, Throwable throwable) {
         logger.error("batch "+batch.getJobBatchNumber()+" failed with error: "+throwable.getMessage());
     }
+
+    protected static String injectConstraintIntoDslQuery(String originalDsl, String constrainingColumn, String constrainingValue) {
+        String constrainingPhrase = ".where(op.gt(op.col(\"" + constrainingColumn + "\"), " + constrainingValue + "))";
+        String constrainedDsl;
+        String regexToRemoveNonQuotedWhitespace = "\\s+(?=(?:[^\\'\"]*[\\'\"][^\\'\"]*[\\'\"])*[^\\'\"]*$)";
+        String originalDslNoWhitespace = originalDsl.replaceAll(regexToRemoveNonQuotedWhitespace, "");
+        if (originalDslNoWhitespace.contains(").")) {
+            int firstClosingParen = originalDslNoWhitespace.indexOf(").");
+            constrainedDsl = originalDslNoWhitespace.substring(0, firstClosingParen+1)
+                + constrainingPhrase + originalDslNoWhitespace.substring(firstClosingParen+1);
+        } else {
+            constrainedDsl = originalDsl + constrainingPhrase;
+        }
+        return constrainedDsl;
+    }
+
+    public String getCurrentQuery() {
```

Comment:
Instead of modifying the interface, what if `newRowBatcher` returned both the RowBatcher and the query? I've used `Pair` in the past for when exactly two things need to be returned, but a custom class - e.g. `RowBatcherContext` - would be useful, particularly when we need to return a third thing. 

I generally prefer that because it a) avoids additional state in the class, b) avoids another method in the interface, and c) most importantly, makes it clear to the reader that the RowBatcher and the query are constructed at the exact same time and are not mutated later on.